File: 프로그래머스/2/42587. 프로세스/프로세스.js
function solution(priorities, location) {
    const queue = [];
    
    for(let i=0;i<priorities.length;i++){
        queue.push({num:i,value:priorities[i]})
    }
    
    const answer = [];
    
    while(queue.length>0){
        const data = queue.shift();
        if(queue.reduce((max, item) => Math.max(max, item.value), 0)>data.value) queue.push(data);
        else answer.push(data);
    }
    console.log(answer);
    
    let result = 0;
    for(let i=0;i<answer.length;i++){
        if(answer[i].num===location) return i+1;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
def solution(priorities, location):
    queue = [(i,p) for i,p in enumerate(priorities)]
    order = 0
        
    while queue:
        value = queue.pop(0)
        
        if any(value[1] < q[1] for q in queue):
            queue.append(value)
        else:
            order += 1
            
            if value[0] == location:
                return order
            